jdaviz/configs/default/plugins/model_fitting/model_fitting.py

```diff
@@ -732,8 +732,8 @@ def _fit_model_to_cube(self, add_data):
         # Retrieve copy of the models with proper "fixed" dictionaries
         models_to_fit = self._reinitialize_with_fixed()
 
-        # Apply masks from selected spectral, spatial subsets
-        self._apply_subset_masks(spec)
+        # Apply masks from selected spatial subsets
+        self._apply_subset_masks(spec, self.spatial_subset)
 
```

Comment:
doesn't the cube fitting case still need to apply both spatial and spectral (or is spectral handled elsewhere here)?

Reply:
I also see 

```
# Reset the data mask in case we use a different subset next time
self._spectrum1d.mask = self._original_mask
```

at the end of the spectral case... I'm guessing we also need that at the end of the cube case to avoid issues when fitting with cube enabled, disabling cube fitting, and attempting to fit the same subset combos to the spectrum.

Reply:
RE first comment: yes, this is needed! Coming in another commit.


RE second comment: In the cube fitting case, the mask gets applied to `spec`:
https://github.com/spacetelescope/jdaviz/blob/41c8235377060cb2b61c06947c68f1cc2a4f7e84/jdaviz/configs/default/plugins/model_fitting/model_fitting.py#L725
which never gets returned to the user and by construction doesn't have a mask to start. So I don't think we need to restore an original mask in that case.

Reply:
ah, I see, spectrum fitting acts on the cached `self._spectrum1d` and cube fitting acts on `spec` which is retrieved on the fly and not cached anywhere (something we might want to refactor to be less confusing eventually, but not necessary in the scope here).